File: DepartmentManagementModels/Employee.cs
using System;

namespace DepartmentManagementModels
{
    public class Employee
    {
        public long EmployeeId { get; set; }
        public string FullName { get; set; }
        public DateTime WasAddedDate { get; set; }
        public DateTime? WasChangedDate { get; set; }
        public DateTime? WasEmployedDate { get; set; }
        public string Position { get; set; }
        public Department Department { get; set; }
    }
}

File: ClientApp/src/app/main/department-management/classes/employee.ts
import {Department} from "./department";

export class Employee {
  employeeId: number = 0;
  fullName: string = '';
  wasAddedDate: Date | null = null;
  wasChangedDate: Date | null = null;
  wasEmployedDate: Date | null = null;
  position: string = '';
  department: Department | null = null;

  constructor(employee: Employee | null = null) {
    if(!employee) return this;
    this.employeeId = employee.employeeId;
    this.fullName = employee.fullName;
    this.wasAddedDate = employee.wasAddedDate;
    this.wasChangedDate = employee.wasChangedDate;
    this.wasEmployedDate = employee.wasEmployedDate;
    this.position = employee.position;
    this.department = employee.department;
  }

}
